From 5ed72af1f6bd0f67b2cc5b87ba2133f537fd8490 Mon Sep 17 00:00:00 2001
From: Jack Jin <1756404649@qq.com>
Date: Thu, 6 Mar 2025 09:23:09 +0800
Subject: [PATCH 3/3] fix(app): update package redirect URLs to include API
 path

- Add '/api/packages/' path to upstream URL for package redirects
- Update redirect logic for package versions and tarballs
- Remove unnecessary resolve() method calls
---
 unpub/lib/src/app.dart | 15 ++++++++-------
 1 file changed, 8 insertions(+), 7 deletions(-)

diff --git a/unpub/lib/src/app.dart b/unpub/lib/src/app.dart
index ca2c7aa..118efd4 100644
--- a/unpub/lib/src/app.dart
+++ b/unpub/lib/src/app.dart
@@ -51,6 +51,7 @@ class App {
     required this.metaStore,
     required this.packageStore,
     // this.upstream = 'https://pub.dev',
+    // this.upstream = 'https://pub.flutter-io.cn',
     this.upstream = 'https://mirrors.tuna.tsinghua.edu.cn/dart-pub',
     this.googleapisProxy,
     this.overrideUploaderEmail,
@@ -157,7 +158,7 @@ class App {
 
     if (package == null) {
       return shelf.Response.found(
-          Uri.parse(upstream).resolve('/api/packages/$name').toString());
+          Uri.parse(upstream + '/api/packages/$name').toString());
     }
 
     package.versions.sort((a, b) {
@@ -187,9 +188,9 @@ class App {
 
     var package = await metaStore.queryPackage(name);
     if (package == null) {
-      return shelf.Response.found(Uri.parse(upstream)
-          .resolve('/api/packages/$name/versions/$version')
-          .toString());
+      return shelf.Response.found(
+          Uri.parse(upstream + '/api/packages/$name/versions/$version')
+              .toString());
     }
 
     var packageVersion =
@@ -206,9 +207,9 @@ class App {
       shelf.Request req, String name, String version) async {
     var package = await metaStore.queryPackage(name);
     if (package == null) {
-      return shelf.Response.found(Uri.parse(upstream)
-          .resolve('/packages/$name/versions/$version.tar.gz')
-          .toString());
+      return shelf.Response.found(
+          Uri.parse(upstream + '/packages/$name/versions/$version.tar.gz')
+              .toString());
     }
 
     if (isPubClient(req)) {
-- 
2.43.0

